File: shared/budgetVersion.js
// Revidert budsjett er vedtatt beregningsgrunnlag for rapportåret 2026.
export const budgetVersionForYear = (year) => String(year) === '2026' ? '2026RV' : `${year}B`;

export const budgetVersionSql = (yearExpression) =>
  `case when cast(${yearExpression} as varchar) = '2026' then '2026RV' else cast(${yearExpression} as varchar) || 'B' end`;

File: shared/budget_version.py
"""Revidert budsjett for 2026; eksisterende budsjettvalg for øvrige år."""


def budget_version_for_year(year: object) -> str:
    return "2026RV" if str(year) == "2026" else f"{year}B"


def budget_version_sql(year_expression: str) -> str:
    return f"case when cast({year_expression} as varchar) = '2026' then '2026RV' else cast({year_expression} as varchar) || 'B' end"
